platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/container/ContainerTransactionGenerator.java

```diff
@@ -13,19 +24,103 @@ public class ContainerTransactionGenerator implements TransactionGenerator {
 
     private static final Logger log = LogManager.getLogger();
 
+    /** Duration between two transaction submissions to a single node. */
+    private static final Duration CYCLE_DURATION = Duration.ofSeconds(1).dividedBy(TPS);
+
+    /** Random instance */
+    private final Random random = new Random();
+
+    /** Supplies the current list of nodes to which transactions should be sent. */
+    private Supplier<List<Node>> nodesSupplier = List::of;
+
+    /** The scheduler used to run the periodic generation job. */
+    private final ScheduledExecutorService scheduler;
+
+    /** The handle of the scheduled generation job returned by the scheduler. */
+    private ScheduledFuture<?> generationTask;
+
+    /** Indicating if the generator is still running */
+    private volatile boolean running = false;
+
+    /**
+     * Creates a new generator with a default single-threaded scheduler.
+     */
+    public ContainerTransactionGenerator() {
+        this(Executors.newSingleThreadScheduledExecutor());
+    }
+
+    /**
+     * Creates a new generator using the provided scheduler. This constructor is primarily intended
+     * for unit tests where a deterministic or mocked scheduler is desirable.
+     *
+     * @param scheduler the scheduler to use for running the generation task
+     */
+    public ContainerTransactionGenerator(final ScheduledExecutorService scheduler) {
+        this.scheduler = scheduler;
+    }
+
+    /**
+     * Sets the supplier that provides the up-to-date list of nodes. Must be called by the network once
+     * the nodes are available.
+     *
+     * @param supplier supplier returning the current nodes
+     */
+    public void setNodesSupplier(final Supplier<List<Node>> supplier) {
+        this.nodesSupplier = supplier;
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public void start() {
-        log.warn("Starting ContainerTransactionGenerator not implemented yet.");
+        if (running) {
+            return;
+        }
+
+        running = true;
+
+        generationTask = scheduler.scheduleAtFixedRate(
+                this::generateAndSubmit, 0, CYCLE_DURATION.toMillis(), TimeUnit.MILLISECONDS);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void stop() {
-        log.warn("Stoping ContainerTransactionGenerator not implemented yet.");
+        running = false;
+
+        if (generationTask != null) {
```

Comment:
It looks like you can remove `running` and use ` generationTask` to check whether the generator is running.